### PR TITLE
Twisted 152 floc 1918

### PR DESCRIPTION
**Pull request by adamtheturtle**
Fixes: https://clusterhq.atlassian.net/browse/FLOC-1918

### CODE REVIEW

File: flocker/common/functional/test_script.py [hunk 1]
```diff
@@ -135,12 +135,10 @@ def test_stdout_stderr(self):
         def got_messages(messages):
             assertContainsFields(self, messages[1],
                                  {u"message_type": u"twisted:log",
-                                  u"message": u"stdout!",
-                                  u"error": False})
+                                  u"message": u"stdout!"})
             assertContainsFields(self, messages[2],
```
Comment (adamtheturtle) [inline flocker/common/functional/test_script.py]: This is the bit I'm not sure about but I'm not sure if this is important. If it is maybe I shouldn't continue with this as it was only meant to be a quick issue to get the test output benefits.

Without this:

```
[FAIL]
Traceback (most recent call last):
  File "/srv/buildslave/flocker-ubuntu-14_04/venv/local/lib/python2.7/site-packages/flocker/common/functional/test_script.py", line 143, in got_messages
    u"error": True})
  File "/srv/buildslave/flocker-ubuntu-14_04/venv/local/lib/python2.7/site-packages/eliot/testing.py", line 44, in assertContainsFields
    test.assertEqual(messageSubset, fields)
  File "/srv/buildslave/flocker-ubuntu-14_04/venv/local/lib/python2.7/site-packages/twisted/trial/_synctest.py", line 437, in assertEqual
    super(_Assertions, self).assertEqual(first, second, msg)
  File "/usr/lib/python2.7/unittest/case.py", line 515, in assertEqual
    assertion_func(first, second, msg=msg)
  File "/usr/lib/python2.7/unittest/case.py", line 837, in assertDictEqual
    self.fail(self._formatMessage(msg, standardMsg))
twisted.trial.unittest.FailTest: {u'message': u'stderr!', u'message_type': u'twisted:log', u'error': False} != {u'message': u'stderr!', u'message_type': u'twisted:log', u'error': True}
- {u'error': False, u'message': u'stderr!', u'message_type': u'twisted:log'}
?            ^^^^

+ {u'error': True, u'message': u'stderr!', u'message_type': u'twisted:log'}
?            ^^^


flocker.common.functional.test_script.FlockerScriptRunnerTests.test_stdout_stderr
```

Comment (exarkun) [inline flocker/common/functional/test_script.py]: I wonder why the existing builder didn't catch this problem?

Comment (exarkun) [inline flocker/common/functional/test_script.py]: Filed this upstream, anyway, https://twistedmatrix.com/trac/ticket/7903

Comment (exarkun) [inline flocker/common/functional/test_script.py]: This probably isn't a difference that really matters to us.  The change in Twisted is a regression, though, so we should probably expect a 15.2.1 that re-introduces the previous behavior.  If that happens we won't want to have changed our tests.

**Pull request closed without merging**